src/revwalk.c

```diff
@@ -83,7 +83,9 @@ int git_revwalk__schedule_commit(git_revwalk *walk, const git_oid *oid, const gi
 
 	commit->uninteresting = opts->uninteresting;
 	list = walk->user_input;
-	if (git_commit_list_insert(commit, &list) == NULL) {
+	if ((opts->insert_by_date &&
+		 git_commit_list_insert_by_date(commit, &list) == NULL) ||
```

Comment:
Sorry to be a nag, but there's another leading space here. While at it, you might also just convert it to use our currently accepted coding style of indenting subsequent if lines by four spaces, only

Reply:
No problem, I forgot the obligatory "correct Xcode indenter" step 😉.